R/utils-get_code_dependency.R

```diff
@@ -33,23 +33,13 @@ get_code_dependency <- function(code, names, check_names = TRUE) {
     return(code)
   }
 
-  # If code is bound in curly brackets, remove them.
-  # TODO: rethink if this is still needed when code is divided by calls?
-  tcode <- trimws(code)
-  if (any(grepl("^\\{.*\\}$", tcode))) {
-    tcode <- sub("^\\{(.*)\\}$", "\\1", tcode)
-  }
-
-  parsed_code <- parse(text = tcode, keep.source = TRUE)
-
-  pd <- utils::getParseData(parsed_code)
-  pd <- normalize_pd(pd)
-  calls_pd <- extract_calls(pd)
-
   if (check_names) {
     # Detect if names are actually in code.
-    symbols <- unlist(lapply(calls_pd, function(call) call[call$token == "SYMBOL", "text"]))
+    parsed_code <- parse(text = trimws(code), keep.source = TRUE)
```

Comment:
Looks like in this if you try to obtain informations which are already in the `lapply(code, attr, "depedency")`

Reply:
Thanks for a fresh look! Will investigate if this could be simplified based on `lapply(code, attr, "depedency")`

Reply:
Simplified in here a8016bc4cf1ee494eae795ef1c722e0db89ed42a